upgrade_test will fail for any stall backtrace

### Prerequisites

* [x] Are you rebased to master ?
* [x] Is it reproducible ?
* [x] Did you perform a cursory search if this issue isn't opened ?

### Versions
* **SCT**: branch-3.1

### Logs
* test_id: 85a67219-cf53-48f8-9429-81d7cd0ecbec
* job: http://jenkins.scylladb.com/view/scylla-3.1/job/scylla-3.1/view/scylla-3.1-test/job/rolling-upgrade/job/rolling-upgrade-centos7/85/execution/node/39/log/
* job log: https://cloudius-jenkins-test.s3.amazonaws.com/85a67219-cf53-48f8-9429-81d7cd0ecbec/job_log.zip

### Description

Some 3.1 upgrade_test failed for finding DB errors (backtrace) in log, but it's just one backtrace from stall (500 ms). 

### Steps to Reproduce

1. run 3.1 upgrade test

**Expected behavior:** We should ignore this kind of backtrace, it won't fail the job.

**Actual behavior:** Job failed.


```
< t:2019-08-26 22:46:08,455 f:tester.py       l:1330 c:sdcm.tester          p:ERROR > Errors found on DB node logs:
< t:2019-08-26 22:46:08,455 f:tester.py       l:1331 c:sdcm.tester          p:ERROR > 2019-08-26 20:41:05.846: (DatabaseLogEvent Severity.ERROR): type=BACKTRACE regex=backtrace line_number=10596 node=Node rolling
-upgrade-3-1-centos-db-node-85a67219-0-4 [35.196.160.56 | 10.142.0.110] (seed: False)
< t:2019-08-26 22:46:08,455 f:tester.py       l:1331 c:sdcm.tester          p:ERROR > 2019-08-26T20:40:51+00:00  rolling-upgrade-3-1-centos-db-node-85a67219-0-4 !INFO    | scylla: Backtrace:
< t:2019-08-26 22:46:08,455 f:tester.py       l:1331 c:sdcm.tester          p:ERROR > 
< t:2019-08-26 22:46:08,455 f:tester.py       l:1331 c:sdcm.tester          p:ERROR > 0x00000000006ab5a2
....

< t:2019-08-26 22:46:08,455 f:tester.py       l:1331 c:sdcm.tester          p:ERROR > 2019-08-26 21:32:13.661: (DatabaseLogEvent Severity.ERROR): type=BACKTRACE regex=backtrace line_number=16029 node=Node rolling
-upgrade-3-1-centos-db-node-85a67219-0-4 [35.196.160.56 | 10.142.0.110] (seed: False)
< t:2019-08-26 22:46:08,455 f:tester.py       l:1331 c:sdcm.tester          p:ERROR > 2019-08-26T21:32:01+00:00  rolling-upgrade-3-1-centos-db-node-85a67219-0-4 !INFO    | scylla: Backtrace:
< t:2019-08-26 22:46:08,455 f:tester.py       l:1331 c:sdcm.tester          p:ERROR > 
....
....
< t:2019-08-26 22:46:08,456 f:tester.py       l:1380 c:sdcm.tester          p:ERROR > Exception in finalize_test method Errors found on DB node logs (see test logs) < t:2019-08-26 22:46:08,456 f:tester.py       l:1380 c:sdcm.tester          p:ERROR > Exception in finalize_test method Errors found on DB node logs (see test logs)
< t:2019-08-26 22:46:08,456 f:tester.py       l:1380 c:sdcm.tester          p:ERROR > Traceback (most recent call last):
< t:2019-08-26 22:46:08,456 f:tester.py       l:1380 c:sdcm.tester          p:ERROR >   File "/sct/sdcm/tester.py", line 1378, in tearDown
< t:2019-08-26 22:46:08,456 f:tester.py       l:1380 c:sdcm.tester          p:ERROR >     self.finalize_test()
< t:2019-08-26 22:46:08,456 f:tester.py       l:1380 c:sdcm.tester          p:ERROR >   File "/sct/sdcm/tester.py", line 1287, in finalize_test
< t:2019-08-26 22:46:08,456 f:tester.py       l:1380 c:sdcm.tester          p:ERROR >     self.stop_resources()
< t:2019-08-26 22:46:08,456 f:tester.py       l:1380 c:sdcm.tester          p:ERROR >   File "/sct/sdcm/tester.py", line 1332, in stop_resources
< t:2019-08-26 22:46:08,456 f:tester.py       l:1380 c:sdcm.tester          p:ERROR >     self.fail('Errors found on DB node logs (see test logs)')
< t:2019-08-26 22:46:08,456 f:tester.py       l:1380 c:sdcm.tester          p:ERROR >   File "/usr/lib64/python2.7/unittest/case.py", line 450, in fail
< t:2019-08-26 22:46:08,456 f:tester.py       l:1380 c:sdcm.tester          p:ERROR >     raise self.failureException(msg)
< t:2019-08-26 22:46:08,456 f:tester.py       l:1380 c:sdcm.tester          p:ERROR > AssertionError: Errors found on DB node logs (see test logs)
```

--- a/sdcm/cluster.py
+++ b/sdcm/cluster.py
@@ -1201,16 +1201,18 @@ class BaseNode(object):
                             self._database_log_errors_index.append(index)
                             e = event.clone_with_info(node=self, line_number=index, line=line)
                             backtraces.append(dict(event=e, backtrace=[]))
+                            matches.append((index, line))
 
-                            if e.severity == Severity.CRITICAL:
-                                matches.append((index, line))
             if not start_from_beginning:
                 self.last_line_no = index if index else last_line_no
                 self.last_log_position = f.tell() + 1
 
         if publish_events:
+            traces_count = 0
             for b in backtraces:
                 b['event'].add_backtrace_info(raw_backtrace="\n".join(b['backtrace']))
+                if b['event'].type == 'BACKTRACE':
+                    traces_count += 1
 
             # filter function to attach the backtrace to the correct error and not to the back traces
             # if the error is within 10 lines and the last isn't backtrace type, the backtrace would be appended to the previous error
@@ -1218,16 +1220,18 @@ class BaseNode(object):
                 last_error = filter_backtraces.last_error
                 try:
                     if (last_error and
-                            backtrace['event'].line_number <= filter_backtraces.last_error.line_number + 10
-                            and not filter_backtraces.last_error.type == 'BACKTRACE'):
+                            backtrace['event'].line_number <= filter_backtraces.last_error.line_number + 20
+                            and not filter_backtraces.last_error.type == 'BACKTRACE' and backtrace['event'].type == 'BACKTRACE'):
                         last_error.add_backtrace_info(raw_backtrace="\n".join(backtrace['backtrace']))
                         return False
                     return True
                 finally:
                     filter_backtraces.last_error = backtrace['event']
-            filter_backtraces.last_error = None
 
-            backtraces = filter(filter_backtraces, backtraces)
+            # support interlaced reactor stalled
+            for _ in range(traces_count):
+                filter_backtraces.last_error = None
+                backtraces = filter(filter_backtraces, backtraces)
 
             for backtrace in backtraces:
                 if Setup.BACKTRACE_DECODING:

--- a/unit_tests/test_cluster.py
+++ b/unit_tests/test_cluster.py
@@ -3,9 +3,11 @@ import tempfile
 import logging
 import shutil
 import os.path
+import json
 
 from sdcm.cluster import BaseNode
 from sdcm.sct_events import start_events_device, stop_events_device
+from sdcm.sct_events import EVENTS_PROCESSES
 
 
 class DummyNode(BaseNode):
@@ -57,23 +59,24 @@ class TestBaseNode(unittest.TestCase):
                              base_logdir=cls.temp_dir, ssh_login_info=dict(key_file='~/.ssh/scylla-test'))
         cls.node.remoter = DummyRemote()
 
-        cls.node.database_log = os.path.join(os.path.dirname(__file__), 'test_data', 'database.log')
-
     @classmethod
     def tearDownClass(cls):
         stop_events_device()
         shutil.rmtree(cls.temp_dir)
 
+    def setUp(self):
+        self.node.database_log = os.path.join(os.path.dirname(__file__), 'test_data', 'database.log')
+
     def test_search_database_log(self):
         critical_errors = self.node.search_database_log()
-        self.assertEqual(len(critical_errors), 4)
+        self.assertEqual(len(critical_errors), 34)
 
         for line_number, line in critical_errors:
             print line
 
     def test_search_database_log_teardown(self):
         critical_errors = self.node.search_database_log(start_from_beginning=True, publish_events=False)
-        self.assertEqual(len(critical_errors), 4)
+        self.assertEqual(len(critical_errors), 36)
 
         for line_number, line in critical_errors:
             print line
@@ -84,3 +87,23 @@ class TestBaseNode(unittest.TestCase):
 
         for line_number, line in errors:
             print line_number, line
+
+    def test_search_database_interlace_reactor_stall(self):
+        self.node.database_log = os.path.join(os.path.dirname(__file__), 'test_data', 'database_interlace_stall.log')
+
+        _ = self.node.search_database_log()
+
+        events_file = open(EVENTS_PROCESSES['MainDevice'].raw_events_filename, 'r')
+        events = []
+        for line in events_file.readlines():
+
+            events.append(json.loads(line))
+
+        event_a, event_b = events[-2], events[-1]
+        print event_a
+        print event_b
+
+        assert event_a["type"] == "REACTOR_STALLED"
+        assert event_a["line_number"] == 1
+        assert event_b["type"] == "REACTOR_STALLED"
+        assert event_b["line_number"] == 4

--- a/unit_tests/test_data/database_interlace_stall.log
+++ b/unit_tests/test_data/database_interlace_stall.log
@@ -1,0 +1,54 @@
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: Reactor stalled for 500 ms on shard 4.
+leapis.com:443 "GET /compute/v1/projects/skilled-adapter-452/zones/us-east1-b/instances/rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 HTTP/1.1" 200 None
+leapis.com:443 "GET /compute/v1/projects/skilled-adapter-452/zones/us-east1-b/instances/rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 HTTP/1.1" 200 None
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: Reactor stalled for 500 ms on shard 4.
+leapis.com:443 "GET /compute/v1/projects/skilled-adapter-452/zones/us-east1-b/instances/rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 HTTP/1.1" 200 None
+leapis.com:443 "GET /compute/v1/projects/skilled-adapter-452/zones/us-east1-b/instances/rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 HTTP/1.1" 200 None
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: Backtrace:
+leapis.com:443 "GET /compute/v1/projects/skilled-adapter-452/zones/us-east1-b/instances/rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 HTTP/1.1" 200 None
+leapis.com:443 "GET /compute/v1/projects/skilled-adapter-452/zones/us-east1-b/instances/rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 HTTP/1.1" 200 None
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x00000000006aa6e2
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x00000000005c1e0c
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x00000000005c22c6
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: /lib64/libpthread.so.0+0x000000000000f5cf
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: /lib64/libc.so.6+0x00000000000c4fac
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: /lib64/libc.so.6+0x00000000000c4e43
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: /lib64/libfreebl3.so+0x0000000000000e21
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: /lib64/libfreebl3.so+0x0000000000000f30
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: Backtrace:
+leapis.com:443 "GET /compute/v1/projects/skilled-adapter-452/zones/us-east1-b/instances/rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 HTTP/1.1" 200 None
+leapis.com:443 "GET /compute/v1/projects/skilled-adapter-452/zones/us-east1-b/instances/rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 HTTP/1.1" 200 None
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x00000000006aa6e2
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x00000000005c1e0c
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x00000000005c22c6
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: /lib64/libpthread.so.0+0x000000000000f5cf
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: /lib64/libc.so.6+0x00000000000c4fac
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: /lib64/libc.so.6+0x00000000000c4e43
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: /lib64/libfreebl3.so+0x0000000000000e21
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: /lib64/libfreebl3.so+0x0000000000000f30
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: /lib64/libcrypt.so.1+0x0000000000002ff9
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: /lib64/libcrypt.so.1+0x0000000000000fcf
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x0000000003ac7dd3
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: /lib64/libcrypt.so.1+0x0000000000002ff9
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: /lib64/libcrypt.so.1+0x0000000000000fcf
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x0000000003ac7dd3
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x0000000003ad27d7
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x0000000003ad27d7
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x0000000003ad304f
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x0000000003ad304f
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x00000000005a31be
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x00000000005a31be
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x00000000005a3381
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x00000000005a3381
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x0000000000671a17
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x0000000000671a17
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x000000000067618a
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x000000000067618a
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x000000000075921d
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: 0x000000000075921d
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: /lib64/libpthread.so.0+0x0000000000007dd4
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: /lib64/libpthread.so.0+0x0000000000007dd4
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: /lib64/libc.so.6+0x00000000000fe02c
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: /lib64/libc.so.6+0x00000000000fe02c
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: [shard 4] schema_tables - Creating keyspace keyspace_entire_test
+00:00  rolling-upgrade-2019-1-n-centos-db-node-b50af2f8-0-3 !INFO    | scylla: [shard 4] schema_tables - Creating keyspace keyspace_entire_test

